### PR TITLE
Enhancement : added all the css prop for the text in the styles section of the Text Widget

### LINKED ISSUE #3391: Feature Request : Add css prop corresponding to text in Text Widget
Environment
Development

Currently, we have a limited number of styles in the Text Widget. We can add more styles such as 
- Font weight: bold, thin, etc + number
- Decoration: normal, underline, strike through
- Style: normal, italic, oblique

### PR DESCRIPTION
**Pull request by manishkushare**
Resolve #3391

### CODE REVIEW

File: frontend/src/Editor/Components/Text.jsx [hunk 1]
```diff
@@ -4,10 +4,32 @@ import DOMPurify from 'dompurify';
...
+    blur,
+    fontVariant,
+  } = styles;
 
+  verticalShadow = `${verticalShadow ?? '0'}px`;
+  horizontalShadow = `${horizontalShadow ?? '0'}px`;
+  blur = `${blur ?? '0'}px`;
   const text = properties.text === 0 || properties.text === false ? properties.text?.toString() : properties.text;
-
+  console.log('shadow', verticalShadow, horizontalShadow, blur, shadowColor);
```
Comment (sherfin94) [inline frontend/src/Editor/Components/Text.jsx:32]: Please remove this console log, and while removing it there are some lint errors that are occurring, please fix those also.

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Small change requested.

**Comment by shubh22** [comment]
Task linked: [CU-2h1aaz2 Adding CSS props to Text Widget](https://app.clickup.com/t/2h1aaz2)

File: frontend/src/Editor/WidgetManager/widgetConfig.js [hunk 2]
```diff
@@ -1589,6 +1589,51 @@ export const widgets = [
...
+          { name: 'normal', value: 'normal' },
+          { name: 'italic', value: 'italic' },
+          { name: 'oblique', value: 'oblique' },
+        ],
+      },
+      lineHeight: { type: 'number', displayName: 'Line Height' },
+      textIndent: { type: 'number', displayName: 'Text Indent' },
+      letterSpacing: { type: 'number', displayName: 'Letter Spacing' },
+      wordSpacing: { type: 'number', displayName: 'Word Spacing' },
+      fontVariant: { type: 'toggle', displayName: 'Font Variant' },
```
Comment (sherfin94) [inline frontend/src/Editor/WidgetManager/widgetConfig.js:1636]: The `font variant` property is not a toggle field, please make it a dropdown.

File: frontend/src/Editor/WidgetManager/widgetConfig.js [hunk 3]
```diff
@@ -1646,12 +1691,20 @@ export const widgets = [
...
         disabledState: { value: '{{false}}' },
+        fontWeight: { value: 'normal' },
+        decoration: { value: 'none' },
+        transformation: { value: 'none' },
+        fontStyle: { value: 'normal' },
+        lineHeight: { value: 1.5 },
+        textIndent: { value: 0 },
+        letterSpacing: { value: 0 },
+        wordSpacing: { value: 0 },
+        fontVariant: { value: '{{false}}' },
```
Comment (sherfin94) [inline frontend/src/Editor/WidgetManager/widgetConfig.js:1707]: Again, not a toggle, so its not a boolean value.

File: server/migrations/1656924847186-addingCssPropsToTextWidget.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            const stylesProps = {
+              fontWeight: 'normal',
+              decoration: 'none',
+              transformation: 'none',
+              fontStyle: 'normal',
+              lineHeight: 1.5,
+              textIndent: 0,
+              letterSpacing: 0,
+              wordSpacing: 0,
+              fontVariant: false,
```
Comment (sherfin94) [inline server/migrations/1656924847186-addingCssPropsToTextWidget.ts:27]: Same issue as described on previous comments.

File: frontend/src/Editor/Components/Text.jsx [hunk 5]
```diff
@@ -31,6 +45,15 @@ export const Text = function Text({ height, properties, styles, darkMode, regist
     display: visibility ? 'flex' : 'none',
     alignItems: 'center',
     textAlign,
+    fontWeight: fontWeight ? fontWeight : fontWeight === '0' ? 0 : 'normal',
+    lineHeight: lineHeight ?? 1.5,
```
Comment (sherfin94) [inline frontend/src/Editor/Components/Text.jsx:49]: Should we also not append `px` for `line height` ?

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Few changes requested.

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**